### PR TITLE
Order scrambles. This fixes #338.

### LINKED ISSUE #338: Order scrambles by schedule
Iskender commented that he could arrange the round order in the old ui to save the time needed to sort the printed sheets.
Once we'll have the schedule in the WCIF, we'll be able to add this neat feature without any additional input from the Delegate!

### PR DESCRIPTION
**Pull request by campos20**
The ui stays as is. The .zip root folder also stays untouched

ordered1

If the schedule is available, TNoodle sorts it considering
Day/Rooms/roundStartTime

and place it on a new folder inside .zip/Printing called **Ordered Scrambles**.

ordered2

If the competition has multiple days, one folder for each is created.

ordered3

Inside the folder Day 1, we have a .pdf for each room

ordered4

If a day has just 1 room, the room name is not stamped 

ordered5

Each pdf is chronologically sorted

Issue: for FM and MBLD, if the attempts aren't placed together, for each attempt, we are printing all the attempts.
Let's say FM attempt 1 starts at 8 a.m., attempt 2 starts at 9:30 and attempt 3 starts at 11 am. We are creating 3 pages for each one, containing all the 3 attempts.

### CODE REVIEW

File: webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java [hunk 1]
```diff
@@ -1608,17 +1618,87 @@ public static ByteArrayOutputStream requestsToZip(ServletContext context, String
...
+                    }
+                    int i = roomList.get(j).indexOf(room);
+                    schedule.get(j).get(i).add(scrambleRequest);
+                }
+            }
+            
+            // This should sort the days correctly
+            // If this is not working, then the folder Day 1 might hold scrambles for another day
+            ArrayList<Integer> temp = new ArrayList<Integer>(dateList);
+            schedule.sort(Comparator.comparingInt(temp::indexOf)); // Java 8
```
Comment (campos20) [inline webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java]: pre-commit, therefore travis, doesn't like this.

File: webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java [hunk 2]
```diff
@@ -1608,17 +1620,92 @@ public static ByteArrayOutputStream requestsToZip(ServletContext context, String
...
+            // let's say attempt = 3 > 1. we are generating 3x more scrambles than needed.
+            for (ScrambleRequest scrambleRequest : scrambleRequests) {
+
+                int attempts = scrambleRequest.scrambles.length;
+//              System.out.println("Attempts: "+attempts);
+
+                Calendar cal = Calendar.getInstance(TimeZone.getTimeZone(scrambleRequest.timeZone));
+                cal.setTime(scrambleRequest.roundStartTime);
+
+                // this will work as long as WCA forbid competitions between years overlap
```
Comment (viroulep) [inline webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java]: FYI there has been "multiyear" competition already, see [Welcome to 2019](https://www.worldcubeassociation.org/competitions/Welcometo2019).

Comment (campos20) [reply webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java]: [Fixed](https://github.com/campos20/tnoodle/commit/e721f21dfefa9ea1247dd2cb3095edcda14c99e4#diff-5c6b26a54fa9a1a113fc535c6c81c91cR1654), thanks!

**Comment by campos20** [comment]
This approach won't work. We are currently [looking at the rounds](https://github.com/thewca/tnoodle/pull/391/files#diff-2afff4f0bf703c666b8af472c7f58fa2R111) and [trying to match](https://github.com/thewca/tnoodle/pull/391/files#diff-2afff4f0bf703c666b8af472c7f58fa2R114) the schedule. This makes us lose information like attempt number, venues, etc. I though of a few hacky solutions (like passing a boolean the says if a round has split attempts), but this would make the code more fuzzy to read. A better approach would be changing how tnoodle generates the scramble to make it more accessible from outside: split functions from [ScrambleRequest](https://github.com/thewca/tnoodle/blob/master/webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java).

**Comment by campos20** [comment]
New approach [here](https://github.com/campos20/tnoodle/tree/order-scrambles-2).

**Pull request closed without merging**